client/packages/programs/src/api/api.ts

```diff
@@ -34,10 +43,39 @@ export const getEncounterQueries = (sdk: Sdk, storeId: string) => ({
     }
     throw new Error('Error querying document');
   },
-  byId: async (encounterId: string): Promise<EncounterFragment> => {
+  byId: async (encounterId: string): Promise<EncounterBaseFragment> => {
     const result = await sdk.encounterById({ encounterId, storeId });
     const encounters = result?.encounters;
 
+    if (
+      encounters?.__typename === 'EncounterConnector' &&
+      !!encounters.nodes[0]
+    ) {
+      return encounters.nodes[0];
+    } else {
+      throw new Error('Could not find encounter');
+    }
+  },
+  previousEncounters: async (
+    patientId: string,
+    current: Date
+  ): Promise<EncounterBaseFragment> => {
+    const result = await sdk.encounters({
+      storeId,
+      key: EncounterSortFieldInput.StartDatetime,
+      desc: true,
+      filter: {
+        startDatetime: {
+          beforeOrEqualTo: new Date(current.getTime() - 1).toISOString(),
```

Comment:
Something for later -- might want to add a strictly `before` filter field so we don't have to worry about the `-1`. 
Not a priority though. :)

Reply:
yes! a `before` and `after` filter is missing